flake.nix

```diff
@@ -6,125 +6,74 @@
 
     systems.url = "github:nix-systems/default";
 
-    terranix-examples.url = "github:terranix/terranix-examples";
-
-    bats-support = {
-      url = "github:bats-core/bats-support";
-      flake = false;
-    };
-
-    bats-assert = {
-      url = "github:bats-core/bats-assert";
-      flake = false;
-    };
-
     flake-parts = {
       url = "github:hercules-ci/flake-parts";
       inputs.nixpkgs-lib.follows = "nixpkgs";
     };
   };
 
-  outputs = inputs @ { self, flake-parts, nixpkgs, ... }:
+  outputs =
+    inputs@{
+      self,
+      flake-parts,
+      nixpkgs,
+      ...
+    }:
 
     flake-parts.lib.mkFlake { inherit inputs; } {
       imports = [
         ./flake-module.nix
+        flake-parts.flakeModules.partitions
       ];
 
-      systems = import
-        inputs.systems;
+      systems = import inputs.systems;
+
+      partitions.dev = {
+        extraInputsFlake = ./dev;
+        module.imports = [ ./dev/flake-module.nix ];
+      };
+
+      partitionedAttrs = {
+        apps = "dev";
+        devShells = "dev";
+        formatter = "dev";
+        templates = "dev";
+      };
```

Comment:
Excellent. I have never tried to use this; I am not in a position to evaluate this yet, but I suppose the effect is obvious (smaller flake lock footprint).

Reply:
Yes, this is what we want to achieve. Since those inputs are used only for testing purposes. With this the downstream users of terranix won't have them added to their flake.lock.